internal/api/registry.go

```diff
@@ -102,10 +109,52 @@ func NewValidator() *validator.Validate {
 type validateContext struct {
 	// Fields must be exported so valdator can access.
 	Method   string
-	Updating bool
 	Resource any
 }
 
-func ValidateRequest(validate *validator.Validate, method string, updating bool, resource any) error {
-	return validate.Struct(validateContext{Method: method, Updating: updating, Resource: resource})
+func ValidateRequest(validate *validator.Validate, method string, resource any) []arm.CloudErrorBody {
```

Comment:
wdyt about unit tests for this?

Reply:
@s-amann I added a new commit for unit tests that cover `required_for_put` validation among other things.  If it's ok with you, I'll defer testing `ValidateRequest` to my preflight branch where the error structure in the response can be even more complicated than what I described in another comment here due to the possibility of multiple resources in the preflight request.